src/main/java/controller/UserController.java

```diff
@@ -0,0 +1,65 @@
+package controller;
+
+import db.DataBase;
+import http.*;
+import model.User;
+import model.Users;
+import template.HandleBars;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Optional;
+
+public class UserController {
+
+    public static void join(HttpRequest httpRequest, HttpResponse httpResponse) {
+        QueryStrings queryStrings = httpRequest.getQueryString();
+
+        User user = new User(queryStrings.getValue("userId"), queryStrings.getValue("password"), queryStrings.getValue("name"), queryStrings.getValue("email"));
+        DataBase.addUser(user);
+
+        httpResponse.response302ToIndexPage();
+    }
+
+    public static void login(HttpRequest httpRequest, HttpResponse httpResponse) {
+        if (isLoginSuccess(httpRequest.getQueryString())) {
+            httpResponse.response302ToLoginSuccessIndexPage();
+        } else {
+            httpResponse.response302ToLoginFailedPage();
+        }
+    }
+
+    private static boolean isLoginSuccess(QueryStrings queryStrings) {
+        String userId = queryStrings.getValue("userId");
+        String password = queryStrings.getValue("password");
+
+        User savedUser = DataBase.findUserById(userId);
+        return savedUser.matchPassword(password);
+    }
+
+    public static void findAll(HttpRequest httpRequest, HttpResponse httpResponse) throws IOException {
+        Optional<String> maybeCookies = httpRequest.getHeaderValue(HeaderName.COOKIE);
```

Comment:
Optional 클래스에서 제공하는 여러가지 기능들이 있는데 이러한 기능을 사용하면 더 간결하게 변경할 수 있을 것 같아요
map, flatMap, filter, ifPresent 등의 기능을 들을 참고해 fluent 사용하는 것을 참고해보세요 
```suggestion
        httpRequest.getHeaderValue(HeaderName.COOKIE)
                .map(Cookies::createCookies)
                .flatMap(cookies -> cookies.getValue("logined"))
                .~~~~
```